plugin-packs/postcss-preset-env/src/options.ts

```diff
@@ -0,0 +1,76 @@
+import type autoprefixer from 'autoprefixer';
+import { pluginsOptions } from './plugins/plugins-options';
+
+export type pluginOptions = {
+	/**
+	 * Determine which CSS features to polyfill,
+	 * based upon their process in becoming web standards.
+	 * default: 2
+	 */
+	stage?: number
+
+	/**
+	 * Determine which CSS features to polyfill,
+	 * based their implementation status.
+	 * default: 0
+	 */
+	minimumVendorImplementations?: number
+
+	/**
+	 * Enable any feature that would need an extra browser library to be loaded into the page for it to work.
+	 * default: false
+	 */
+	enableClientSidePolyfills?: boolean
+
+	/**
+	 * PostCSS Preset Env supports any standard browserslist configuration,
+	 * which can be a `.browserslistrc` file,
+	 * a `browserslist` key in `package.json`,
+	 * or `browserslist` environment variables.
+	 *
+	 * The `browsers` option should only be used when a standard browserslist configuration is not available.
+	 */
+	browsers?: string | readonly string[] | null
+
+	/**
+	 * Determine whether all plugins should receive a `preserve` option,
+	 * which may preserve or remove the original and now polyfilled CSS.
+	 * Each plugin has it's own default, some true, others false.
+	 * default: _not set_
+	 */
+	preserve?: boolean
+
+	/**
+	 * [Configure autoprefixer](https://github.com/postcss/autoprefixer#options)
+	 */
+	autoprefixer?: autoprefixer.Options
+
+	/**
+	 * Enable or disable specific polyfills by ID.
+	 * Passing `true` to a specific [feature ID](https://github.com/csstools/postcss-plugins/blob/main/plugin-packs/postcss-preset-env/FEATURES.md) will enable its polyfill,
+	 * while passing `false` will disable it.
+	 *
+	 * Passing an object to a specific feature ID will both enable and configure it.
+	 */
+	features?: pluginsOptions
+
+	/**
+	 * The `insertBefore` key allows you to insert other PostCSS plugins into the chain.
+	 * This is only useful if you are also using sugary PostCSS plugins that must execute before certain polyfills.
+	 * `insertBefore` supports chaining one or multiple plugins.
+	*/
+	insertBefore?: Record<string, any> // eslint-disable-line @typescript-eslint/no-explicit-any
+
+	/**
+	 * The `insertAfter` key allows you to insert other PostCSS plugins into the chain.
+	 * This is only useful if you are also using sugary PostCSS plugins that must execute after certain polyfills.
+	 * `insertAfter` supports chaining one or multiple plugins.
+	*/
+	insertAfter?: Record<string, any> // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
Agreed, this could be a bad comment since I have limited TS knowledge but should we use `unknown` instead of `any`?

Reply:
`unknown` is better.
If I understand it correctly it doesn't make a difference for users of the plugin because they can assign anything to both `any` and `unknown`.

But `unknown` will warn us when we add more typings to `postcss-preset-env` internals, `any` will not.

So `unknown` should be better and I've changed it to that :)